### PR TITLE
ddl, cmd: reorganize test cases in 'cmd/ddltest', delete `TestCommitWhenSchemaChanged` case, move `TestForIssue24621` to `ddl/column_type_change_test.go`

### LINKED ISSUE #28435: cmd/ddltest TestDDLSuiteTestForIssue24621 doesn't run
This test was introduced by https://github.com/pingcap/tidb/pull/27556 .

But it is _not_ triggered by check_dev_2 or [integration_ddl_test](https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tidb/tidb_ghpr_integration_ddl_test.groovy).

I create https://github.com/pingcap/tidb/pull/28433 to verify my statement.

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #28435

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Defined2014** [comment]
/cc @tisonkun @zimulala @xiongjiwei @xhebox 
PTAL

File: ddl/column_type_change_test.go [hunk 1]
```diff
@@ -2227,3 +2227,14 @@ func (s *testColumnTypeChangeSuite) TestChangeFromBitToStringInvalidUtf8ErrMsg(c
 	errMsg := "[table:1366]Incorrect string value '\\xEC\\xBD' for column 'a'"
 	tk.MustGetErrMsg("alter table t modify column a varchar(31) collate utf8mb4_general_ci;", errMsg)
 }
+
+func (s *testColumnTypeChangeSuite) TestForIssue24621(c *C) {
```
Comment (Defined2014) [inline ddl/column_type_change_test.go:2231]: Do we need this test anymore? I think it's already coverd by others tests.

Comment (zimulala) [reply ddl/column_type_change_test.go:2231]: We can delete this test if we already have one. But would it be nice to say where this case is already covered?

**Review by zimulala — COMMENTED** [review]
According to the related issue, that whether we also need to do the processing of such other tests, such as TestCommitWhenSchemaChanged?

File: ddl/column_type_change_test.go (hunk 1 above)
Comment (Defined2014) [reply ddl/column_type_change_test.go:2231]: A bit like, but not exactly the same. Maybe the best way is to keep it.

**Comment by Defined2014** [comment]
> According to the related issue, that whether we also need to do the processing of such other tests, such as TestCommitWhenSchemaChanged?

Very similar to the test case below. Delete it firstly. https://github.com/pingcap/tidb/blob/b1352ded030c33065e711440939b97767a518d76/session/session_test.go#L2321-L2336

**Comment by Defined2014** [comment]
After this PR, ddltest package has the test cases below
```
TestSimple
TestSimpleInsert
TestSimpleUpdate
TestSimpleDelete
TestIndex
TestColumn
```

All be covered by IT test.

**Review by zimulala — APPROVED** [review]
Please add the operations(delete a test and move a test) in the description

LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**